catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -810,6 +878,41 @@ public CreateResponse create(CreateRequest createRequest)
 
         Exception ingestError = null;
         try {
+            if (policy != null) {
```

Comment:
I've reduced how big these sections are to make it a bit more readable. I'm going to leave the major refactoring up to a different ticket, however.